src/components/customerPortal/usage/UsageSubscriptionItem.tsx

```diff
@@ -0,0 +1,110 @@
+import { LoaderUsageSubscriptionItem } from '~/components/customerPortal/common/SectionLoading'
+import { planRenewalDate } from '~/components/customerPortal/utils'
+import { intlFormatNumber } from '~/core/formats/intlFormatNumber'
+import { deserializeAmount } from '~/core/serializers/serializeAmount'
+import { PlanInterval, SubscriptionForPortalUsageFragment, TimezoneEnum } from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+
+type UsageSubscriptionItemProps = {
+  subscription?: SubscriptionForPortalUsageFragment | null
+  applicableTimezone?: TimezoneEnum | null
+  loading?: boolean
+  children?: React.ReactNode
+}
+
+const planIntervalLabel = (interval?: string | null) => {
```

Comment:
Nitpicking, seems interval will always be present as you have a condition above invocation.

Also, maybe a lookup object could help being more concise as you did for `INVOICE_TYPE_TRANSLATION_MAP`!

Reply:
Good point, removed the optional part and created a map.